### PR TITLE
[SPARK-47370][Doc] Add migration doc: TimestampNTZ type inference on Parquet files

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Add migration doc: TimestampNTZ type inference on Parquet files

### Why are the changes needed?

Update docs. The behavior change was not mentioned in the SQL migration guide

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

It's just doc change
### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ license: |
   - Since Spark 3.4, vectorized readers are enabled by default for the nested data types (array, map and struct). To restore the legacy behavior, set `spark.sql.orc.enableNestedColumnVectorizedReader` and `spark.sql.parquet.enableNestedColumnVectorizedReader` to `false`.
   - Since Spark 3.4, `BinaryType` is not supported in CSV datasource. In Spark 3.3 or earlier, users can write binary columns in CSV datasource, but the output content in CSV files is `Object.toString()` which is meaningless; meanwhile, if users read CSV tables with binary columns, Spark will throw an `Unsupported type: binary` exception.
   - Since Spark 3.4, bloom filter joins are enabled by default. To restore the legacy behavior, set `spark.sql.optimizer.runtime.bloomFilter.enabled` to `false`.
+  - Since Spark 3.4, when schema inference on external Parquet files, INT64 timestamps with annotation `isAdjustedToUTC=false` will be inferred as TimestampNTZ type instead of Timestamp type. To restore the legacy behavior, set `spark.sql.parquet.inferTimestampNTZ.enabled` to `false`.
```
Comment (dongjoon-hyun) [inline docs/sql-migration-guide.md:70]: SPARK-38829 seems to add this at 3.3.0. When this behavior change happens, @gengliangwang ?

Comment (gengliangwang) [reply docs/sql-migration-guide.md:70]: The TimestampNTZ feature was disabled in Spark 3.3 (https://issues.apache.org/jira/browse/SPARK-38813) and release in Spark 3.4

Comment (dongjoon-hyun) [reply docs/sql-migration-guide.md:70]: Thank you for the confirmation.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by gengliangwang** [comment]
@dongjoon-hyun Thanks, merging to master/branch-3.5/branch-3.4

**Pull request closed without merging**